<launch>
   
  <!-- VEHICLE FRAMES -->
  <rosparam command="load" file="$(find xiroi)/config/xiroi_frames.yaml"/>

  <!-- NED Origin -->
  <arg name="place" default="emissari.yaml"/>
  <arg name="placesDir" default="$(find xiroi)/config/places"/>
  <rosparam command="load" file="$(arg placesDir)/$(arg place)"/>  
  <node pkg="tf" type="static_transform_publisher" name="utm2map" args="0 0 0 0 0 0 1 utm map 100"/>

  <!--DYNAMICS -->
  <rosparam command="load" file="$(find xiroi)/config/robot_dynamics.yaml"/>
  <node name="robot_dynamics" pkg="xiroi" type="robot_dynamics.py" respawn="false" output="screen">
     <remap from="/control/thrusters_data" to="/setpoints"/>
  </node>

  <!--SENSORS-->
  <!--IMU:publishes /sensors/imu-->
  <include file="$(find xiroi)/launch/imu.launch"/>
  <!--EMLID GPSpublishes /sensors/gps-->
  <!-- <include file="$(find emlid_reach_tcp)/launch/emlid_gps.launch"/> -->
  <!--USBL-->
  <!-- <include file="$(find evologics_ros_sync)/launch/evo_gs.launch"/> -->

   <!--OPERATION NODE-->
  <!-- <node name="operator" pkg="xiroi" type="operator.py" respawn="false" output="screen"/> -->

  <!--ROBOT LOCALIZATION-->
  <!-- <rosparam command="load" file="$(find xiroi)/config/robot_localization_operation.yaml" /> -->
  <!--continuous data sensors odom->xiroi-->
 <!--  <node pkg="robot_localization" type="ekf_localization_node" name="ekf_odom" clear_params="true">
    <remap from="odometry/filtered" to="odometry/filtered_odom"/>
  </node> -->
  <!--continuous data sensors + GPS  map->odom-->
 <!--  <node pkg="robot_localization" type="ekf_localization_node" name="ekf_map" clear_params="true">
    <remap from="odometry/filtered" to="odometry/filtered_map"/>
  </node> -->
  <!--navsat_transform_node-->
<!--  <node pkg="robot_localization" type="navsat_transform_node" name="navsat_transform" respawn="true"> 
    <param name="magnetic_declination_radians" value="0"/>
    <param name="yaw_offset" value="0"/>
    <remap from="/imu/data" to="/sensors/imu" />
    <remap from="/gps/fix" to="/sensors/gps" />
    <remap from="/odometry/filtered" to="/odometry/filtered_odom" /> 
    <remap from="/odometry/gps" to="/sensors/gps_odom"/>
   </node> -->
 
   
   
</launch>